packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -61,21 +119,103 @@ public function createIndex(int $domainId, string $index)
             'index' => $indexName,
             'body' => $definition,
         ]);
+        if ($setToSettings) {
+            $this->setting->setForDomain(Setting::ELASTICSEARCH_INDEX, $indexName, $domainId);
```

Comment:
This means when I `createIndex`, then this new index name will be set as  "the index" for the domain? This is quite strange functionality because it combines 2 things - create index and set the index as "the index" for the domain.

I suggest to create a new method that will be responsible for this setting, and I have a feeling that this method shouldn't be part of `ElasticsearchStructureManager`

Reply:
Even after discussions I suggest to remove this `setToSettings` from the function. You can create a new function 

```php
public function createIndexAndForDomain(int $domainId, string $index)
{
 $this->createIndex($domainId, $index);
 $indexName = $this->getIndexName($domainId, $index);
 $this->setting->setForDomain(Setting::ELASTICSEARCH_INDEX, $indexName, $domainId);
}
```